pkg/cmd/server/origin/master.go

```diff
@@ -314,7 +319,13 @@ func (c *MasterConfig) buildHandlerChain(assetConfig *AssetConfig) (func(http.Ha
 
 		handler = kgenericfilters.WithCORS(handler, c.Options.CORSAllowedOrigins, nil, nil, nil, "true")
 		handler = kgenericfilters.WithPanicRecovery(handler, c.RequestContextMapper)
-		handler = kgenericfilters.WithTimeoutForNonLongRunningRequests(handler, kc.LongRunningFunc)
+
+		longRunningRequestRE := regexp.MustCompile(longRunningEndpointsRE)
+		longRunningRequestCheck := kgenericfilters.BasicLongRunningRequestCheck(longRunningRequestRE, nil)
+		longRunningRequestCheckFunc := func(r *http.Request) bool {
+			return longRunningRequestCheck(r) || kc.LongRunningFunc(r)
+		}
+		handler = kgenericfilters.WithTimeoutForNonLongRunningRequests(handler, longRunningRequestCheckFunc)
```

Comment:
It complicates the handler chain even more. There were complaints already in this direction. Moreover, the `Config.LongRunningFunc` exists for the very reason of customizability. Otherwise, we would have hardcoded it in the upstream code.